test/EndToEnd/ProjectBuildTests.cs

```diff
@@ -248,6 +248,29 @@ public void ItCanCreateItemTemplate(string templateName)
             Assert.True(directory.EnumerateFileSystemInfos().Any());
         }
 
+        [Theory]
+        // microsoft.dotnet.common.itemtemplates templates
+        [InlineData("class")]
+        [InlineData("struct")]
+        [InlineData("enum")]
+        [InlineData("record")]
+        [InlineData("interface")]
+        public void ItCanCreateItemTemplateWithProjectRestriction(string templateName, string language = "C#")
+        {
+            DirectoryInfo directory = InstantiateProjectTemplate("classlib", language, withNoRestore: false);
+            string projectDirectory = directory.FullName;
+            string newArgs = $"{templateName} --language {language} --debug:ephemeral-hive";
+
+            new NewCommandShim()
+                .WithWorkingDirectory(projectDirectory)
+                .Execute(newArgs)
+                .Should().Pass();
+
+            //check if the template created files
+            Assert.True(directory.Exists);
+            Assert.True(directory.EnumerateFileSystemInfos().Any());
```

Comment:
this check is a bit weak, as project instantiation also creates the files.
it would be good to check that expected file for item template was created.

Reply:
done